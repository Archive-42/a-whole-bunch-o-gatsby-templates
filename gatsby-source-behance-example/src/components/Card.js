import React from 'react';
import { FaThumbsUp, FaEye, FaComment } from 'react-icons/fa';
import format from 'date-fns/format';
import parse from 'date-fns/parse';
import css from './Card.module.css';

const Card = props => (
  <div className="col card">
    <a
      href={props.url}
      target="_blank"
      rel="noopener noreferrer"
      className={css.wrapper}
      style={{ backgroundImage: `url(${props.cover})` }}
    >
      <div className={css.header}>
        <div className={css.date}>
          <Date date={props.date} />
        </div>
        <div className={css.stats}>
          <div className={css.views}>
            <FaEye />
            <span>{props.views}</span>
          </div>
          <div className={css.appreciations}>
            <FaThumbsUp />
            <span>{props.appreciations}</span>
          </div>
          <div className={css.comments}>
            <FaComment />
            <span>{props.comments}</span>
          </div>
        </div>
      </div>
      <div className={css.data}>
        <div className={css.content}>
          <span className={css.areas}>
            {props.areas.map(area => (
              <span key={area} className={css.areasItem}>
                {area}
              </span>
            ))}
          </span>
          <h2>{props.name}</h2>
        </div>
      </div>
    </a>
  </div>
);

export default Card;

const Date = ({ date }) => {
  const parsed = parse(date * 1000).toString();
  const formatted = format(parsed, 'DD.MM.YYYY');

  return <span>{formatted}</span>;
};
