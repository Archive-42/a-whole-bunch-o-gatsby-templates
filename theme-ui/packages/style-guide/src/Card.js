/** @jsx jsx */
import { jsx } from 'theme-ui'

export const Card = props => (
  <div
    {...props}
    sx={{
      variant: 'styles.Card',
    }}
  />
)

export default Card
