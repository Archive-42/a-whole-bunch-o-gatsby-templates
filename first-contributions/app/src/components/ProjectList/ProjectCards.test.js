import React from 'react';
import renderer from 'react-test-renderer';
import Card from './ProjectsCards';

it('renders a project card', () => {
  const tree = renderer.create(
    <Card
              name='Awesome Project'
              logoLink='http://awesome-project.com/logo.jpg'
              githubLink='https://github.com/awesome-project'
              description='Simply awesome Project'
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

