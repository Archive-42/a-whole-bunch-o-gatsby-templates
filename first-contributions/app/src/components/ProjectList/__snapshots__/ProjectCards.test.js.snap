// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a project card 1`] = `
<div
  className="Card-Container"
>
  <a
    className="Card-Real-Link"
    href="https://github.com/awesome-project"
  >
    <div
      className="Card-Header"
    >
      <img
        alt="the framework or language that the project is build upon"
        className="Project-Logo"
        src="http://awesome-project.com/logo.jpg"
      />
      <h3
        className="Card-Title"
      >
        Awesome Project
      </h3>
    </div>
    <div
      className="Card-Body"
    >
      <div
        className="Card-Description"
      >
        <p>
           
          Simply awesome Project
        </p>
      </div>
      <div
        className="Card-Link"
      >
        Go to Project
      </div>
    </div>
  </a>
</div>
`;
